service/src/test/java/uk/gov/hmcts/reform/fpl/service/docmosis/EPOGenerationServiceTest.java

```diff
@@ -60,6 +69,64 @@ void shouldGetTemplateDataWhenGivenPopulatedCaseDataInDraft() {
         assertThat(templateData).isEqualToComparingFieldByField(expectedData);
     }
 
+    @Test
+    void shouldVerifyExclusionRequirementWhenTheStartDateIsSameAndToggledOn() {
+        given(featureToggleService.isEpoOrderTypeAndExclusionEnabled()).willReturn(true);
+        CaseData caseData = getEpoExclusionRequirementCase(DRAFT,
+            EPOExclusionRequirementType.STARTING_ON_SAME_DATE,
+            LocalDate.of(2021, 1, 13),
+            "Test User");
+
+        DocmosisGeneratedOrder templateData = service.populateCustomOrderFields(caseData);
+
+        assertThat(templateData.getExclusionRequirement()).isEqualTo("The Court directs that Test User be excluded"
+            + " from 1 Main Street, Lurgan, BT66 7PP, Armagh, United Kingdom forthwith so that the child may "
+            + "continue to live there, consent to the exclusion requirement having been given by Test User.");
+    }
+
+    @Test
+    void shouldVerifyExclusionRequirementWhenTheStartDateIsDifferentAndToggledOn() {
+        given(featureToggleService.isEpoOrderTypeAndExclusionEnabled()).willReturn(true);
+        CaseData caseData = getEpoExclusionRequirementCase(DRAFT,
+            EPOExclusionRequirementType.STARTING_ON_DIFFERENT_DATE,
+            LocalDate.of(2021, 1, 26),
+            "Test User");
+
+
+        DocmosisGeneratedOrder templateData = service.populateCustomOrderFields(caseData);
+
+        assertThat(templateData.getExclusionRequirement()).isEqualTo("The Court directs that Test User be excluded"
+            + " from 1 Main Street, Lurgan, BT66 7PP, Armagh, United Kingdom "
+            + "from 26 January 2021 so that the child may "
+            + "continue to live there, consent to the exclusion requirement having been given by Test User.");
+    }
+
+    @Test
+    void shouldVerifyExclusionRequirementWhenNoToExclusionHasBeenSelectedAndToggledOn() {
+        given(featureToggleService.isEpoOrderTypeAndExclusionEnabled()).willReturn(true);
+        CaseData caseData = getEpoExclusionRequirementCase(DRAFT,
+            EPOExclusionRequirementType.NO_TO_EXCLUSION,
+            LocalDate.of(2021, 1, 13),
+            "Temp User");
+
+        DocmosisGeneratedOrder templateData = service.populateCustomOrderFields(caseData);
+
+        assertThat(templateData.getExclusionRequirement()).isEqualTo(null);
+    }
+
+    @Test
+    void shouldVerifyExclusionRequirementIsNullWhenToggledOff() {
+        given(featureToggleService.isEpoOrderTypeAndExclusionEnabled()).willReturn(false);
+        CaseData caseData = getEpoExclusionRequirementCase(DRAFT,
+            EPOExclusionRequirementType.STARTING_ON_DIFFERENT_DATE,
+            LocalDate.of(2021, 1, 13),
+            "Some User");
+
+        DocmosisGeneratedOrder templateData = service.populateCustomOrderFields(caseData);
+
+        assertThat(templateData.getExclusionRequirement()).isEqualTo(null);
```

Comment:
assertThat(templateData.getExclusionRequirement()).isNull;

can use the above instead ^^ code smell shown in sonar

Reply:
sorted